File: src/main/java/org/example/kalkulationsprogramm/repository/WebsiteAnalyticsSnapshotRepository.java
package org.example.kalkulationsprogramm.repository;

import org.example.kalkulationsprogramm.domain.WebsiteAnalyticsSnapshot;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.Optional;

public interface WebsiteAnalyticsSnapshotRepository extends JpaRepository<WebsiteAnalyticsSnapshot, Long> {

    Optional<WebsiteAnalyticsSnapshot> findBySnapshotDate(LocalDate snapshotDate);

    Optional<WebsiteAnalyticsSnapshot> findFirstByOrderBySnapshotDateDesc();
}

File: src/main/kotlin/org/example/kalkulationsprogramm/repository/WebsiteAnalyticsSnapshotRepository.kt
package org.example.kalkulationsprogramm.repository

import java.time.LocalDate
import java.util.Optional
import org.example.kalkulationsprogramm.domain.WebsiteAnalyticsSnapshot
import org.springframework.data.jpa.repository.JpaRepository

interface WebsiteAnalyticsSnapshotRepository : JpaRepository<WebsiteAnalyticsSnapshot, Long> {
    fun findBySnapshotDate(snapshotDate: LocalDate): Optional<WebsiteAnalyticsSnapshot>

    fun findFirstByOrderBySnapshotDateDesc(): Optional<WebsiteAnalyticsSnapshot>
}
